bridgestyle/sld/fromgeostyler.py

```diff
@@ -27,32 +59,30 @@ def convert(geostyler):
     userStyleTitle = SubElement(userStyle, "Title")
     userStyleTitle.text = geostyler["name"]
 
-    featureTypeStyle = SubElement(userStyle, "FeatureTypeStyle")
-    if "transformation" in geostyler:
-        featureTypeStyle.append(processTransformation(geostyler["transformation"]))
-    for rule in geostyler.get("rules", []):
-        featureTypeStyle.append(processRule(rule))
-    if "blendMode" in geostyler:
-        _addVendorOption(featureTypeStyle, "composite", geostyler["blendMode"])
+    z_values = list(rulesByZ.keys())
+    z_values.sort()
+    for z_value in z_values:
+        zrules = rulesByZ[z_value]
+        featureTypeStyle = SubElement(userStyle, "FeatureTypeStyle")
+        if "transformation" in geostyler:
+            featureTypeStyle.append(processTransformation(geostyler["transformation"]))
+        for rule in zrules:
+            featureTypeStyle.append(processRule(rule))
+        if "blendMode" in geostyler:
+            _addVendorOption(featureTypeStyle, "composite", geostyler["blendMode"])
         
     sldstring = ElementTree.tostring(root, encoding='utf8', method='xml').decode()
-    dom = minidom.parseString(sldstring)    
-    return dom.toprettyxml(indent="  "), _warnings
+    dom = minidom.parseString(sldstring)
+    result = dom.toprettyxml(indent="  "), _warnings
+    return result
```

Comment:
The result is expected to be a tuple, not just the xml

Reply:
It does return the tuple - just as before.  I just put the result in a variable so I could see it more easily in the debugger.